controllers/logging/logging_controller.go

```diff
@@ -298,11 +298,8 @@ func (r *LoggingReconciler) dynamicDefaults(ctx context.Context, log logr.Logger
 	if err := r.Client.List(ctx, &nodes); err != nil {
 		log.Error(err, "listing nodes")
 	}
-	if logging.Spec.SyslogNGSpec != nil && len(nodes.Items) > 0 {
-		logging.Spec.SyslogNGSpec.MaxConnections = len(nodes.Items) * 10
-		if logging.Spec.SyslogNGSpec.MaxConnections > 512 {
-			logging.Spec.SyslogNGSpec.MaxConnections = 512
-		}
+	if logging.Spec.SyslogNGSpec != nil && logging.Spec.SyslogNGSpec.MaxConnections == 0 {
+		logging.Spec.SyslogNGSpec.MaxConnections = max(100, min(1000, len(nodes.Items)*10))
```

Comment:
Not sure about the intention behind this one: The upper bound is 100, but the inner `min` checks for 1000. Maybe the values are swapped?

Reply:
We had an IRL conversation about this, long story short the upper bound 100 is the important part here.